doc/user-guide/data_model.rst

```diff
@@ -0,0 +1,251 @@
+
+* Wrapping a VTK array
+* Using a :class:`numpy.ndarray` array
+* Or just using a :class:`list`
+
+PyVista provides pythonic methods for all three approaches so you can
+choose whatever is most efficient for you.  If you're comfortable with
+the VTK API, you can choose to wrap VTK arrays, but you may find that
+using :class:`numpy.ndarray` is more convienent and avoids the looping
+overhead in Python.
+
+Wrapping a VTK Array
+~~~~~~~~~~~~~~~~~~~~
+Let's define points of a triangle.  Using the VTK API, this can be
+done with:
+
+.. jupyter-execute::
+
+   >>> import vtk
+   >>> vtk_array = vtk.vtkDoubleArray()
+   >>> vtk_array.SetNumberOfComponents(3)
+   >>> vtk_array.SetNumberOfValues(9)
+   >>> vtk_array.SetValue(0, 0)
+   >>> vtk_array.SetValue(1, 0)
+   >>> vtk_array.SetValue(2, 0)
+   >>> vtk_array.SetValue(3, 1)
+   >>> vtk_array.SetValue(4, 0)
+   >>> vtk_array.SetValue(5, 0)
+   >>> vtk_array.SetValue(6, 0.5)
+   >>> vtk_array.SetValue(7, 0.667)
+   >>> vtk_array.SetValue(8, 0)
+   >>> print(vtk_array)
+
+PyVista supports creating objects directly from vtkDataArrays, but
+there's a better, and more pythonic alternative by using
+:class:`numpy.ndarray`.
+
+Using NumPy with PyVista
+~~~~~~~~~~~~~~~~~~~~~~~~
+However, there's no reason to do this since Python already has the
+excellent C array library `NumPy <https://numpy.org/>`_.  You could
+more create a points array with:
+
+.. jupyter-execute::
+
+   >>> import numpy as np
+   >>> np_points = np.array([[0, 0, 0],
+   ...                       [1, 0, 0],
+   ...                       [0.5, 0.667, 0]])
+   >>> np_points
+
+We use a :class:`numpy.ndarray` here so that PyVista directly "point"
+the underlying C array to VTK.  VTK already has APIs to directly read
+in the C arrays from ``numpy``, and since VTK is written in C++,
+everything from Python that is transferred over to VTK needs to be in a
+format that VTK can process.
+
+Should you wish to use VTK objects within PyVista, you can still do
+this.  In fact, using :func:`pyvista.wrap`, you can even get a numpy-like
+representation of the data.  For example:
+
+.. jupyter-execute::
+
+   >>> import pyvista
+   >>> wrapped = pyvista.wrap(vtk_array)
+   >>> wrapped
+
+Note that when wrapping the underlying VTK array, we actually perform
+a shallow copy of the data.  In other words, we pass the pointer from
+the underlying C array to the numpy :class:`numpy.ndarray`, meaning
+that the two arrays are now efficiently linked.  This means that we
+can change the array using numpy array indexing and have it modified
+on the "VTK side".
+
+.. jupyter-execute::
+
+   >>> wrapped[0, 0] = 10
+   >>> vtk_array.GetValue(0)
+
+Or we can change the value from the VTK array and see it reflected in
+the numpy wrapped array.  Let's change the value back:
+
+.. jupyter-execute::
+
+   >>> vtk_array.SetValue(0, 0)
+   >>> wrapped[0, 0]
+
+
+Using a Python List
+~~~~~~~~~~~~~~~~~~~
+PyVista supports the use of Python lists, and you could define a your
```

Comment:
If we support any sequence (which we probably do), we should probably say so. Perhaps only in parentheses, not to scare off users who don't know what that means, but this information could still benefit more savvy users (by noting that they don't have to transform their tuples into lists, for instance).

Reply:
@adeak Agreed. both `lists` and `tuples` (and numpy arrays made of `lists` and `tuples` are accepted). Could we be explicit and say "sequences (i.e. lists or tuples)" for the less-initiated pythonistas? :)

Reply:
Good points.  Added this.